### ISSUE TITLE
decompress 1.3.0 tests fail to build due to bigarry pulling in dune 

### ISSUE THREAD

**Issue by sternenseemann** [mirage/decompress#113]
When I build the tests with NixOS, the following happens. Reproducible with 4.10, 4.11 and 4.12.

```
File "bindings/stubs/dune", line 15, characters 22-26:
15 |  (forbidden_libraries unix)
                           ^^^^
Error: Library "unix" was pulled in.
-> required by library "bigarray" in
   /nix/store/56bmf5x8l90znfdvigxbj92iwzdyrva9-ocaml-4.11.2/lib/ocaml
-> required by library "ctypes" in
   /nix/store/l91x7cq89449y7xcj1sq8645nkkmh6vh-ocaml4.11.2-ctypes-0.17.1/lib/ocaml/4.11.2/site-lib/ctypes
-> required by library "ctypes.stubs" in
   /nix/store/l91x7cq89449y7xcj1sq8645nkkmh6vh-ocaml4.11.2-ctypes-0.17.1/lib/ocaml/4.11.2/site-lib/ctypes
-> required by library "gen_decompress_bindings" in
   _build/default/bindings/stubs
        test alias test/runtest
```

Patching this line out makes the tests pass without a hitch. Is this safe to do?

**Comment by dinosaure** [comment]
Hmmhmm, it's intentional to forbidden `unix` and `coupes.0.18.0` should **not** depends on `unix`. I'm not sure about your context but it seems that we have linking tricks. So I would like to say that it's safe but not expected 😕

**Comment by sternenseemann** [comment]
Found the issue which was related to some linker shenanigans indeed: For some reason having `ctypes` 0.17.1 in the environment made something pull in `unix`, the trace lists `ctypes` which got me suspicious, but it seems misleading, as bigarray shouldn't be the cause?

Anyways, you probably know this better than me as you are the author of the switch away from `unix` in `ctypes` :)

tl;dr: updating ctypes to 0.18.0 fixes my issue.
